### DISCUSSION TITLE
Allow Snakes to have multiple colors/customizations

### DISCUSSION THREAD

**Discussion by BoldBigflank** [BattlesnakeOfficial/feedback#193] [Feature Requests]
I would like to have my snake be multiple different colors for aesthetic purposes. There are a few ways we could make this happen.

* Include snake id as a query param in the GET /info route
If instead of just calling `https://{battlesnake_url}/info` it could call `https://{battlesnake_url}/info?id=snk_XXXXXXXXXXXXXXXXX` and then my logic could change the response data based on that query data. It wouldn't break existing snakes, but I would need to look up colors either in code or a database to take advantage of it.

* Allow for including a custom query param in all requests
In the "Edit Snake" page, perhaps have an option of a query string that will be passed to all requests. This is similar to the first, but different enough to be its own idea.
![image](https://user-images.githubusercontent.com/636964/161198718-0a29a4bb-3975-47cc-9885-431c3c5dcc4f.png)

* Let more advanced users set the url for each option:
While the "Battlesnake URL" field makes a bunch of assumptions, you could also have a "custom endpoints" checkbox that opens up four more options, where you could set the full url for each endpoint.

![image](https://user-images.githubusercontent.com/636964/161197960-af59808c-0069-4576-9f1e-7f848cc6011a.png)

**Comment by BoldBigflank** [comment]
One other idea: Let the /info response color field be an array

```
{
        apiversion: "1",
        author: "boldbigflank",
        color: ["#1778B5", "#ff00ff", "#BAD455"],
        head: "orca",
        tail: "shiny",
        version: '1.0.0'
    }
```
Then the game could either be set up to randomly pick a color each time, or only use the second and third colors when there are multiple snakes of the same id.

**Comment by robbles** [comment]
Hey @BoldBigflank, thanks for the suggestion!
If I'm understanding what you're asking for correctly, you can already do something like this with the URL field for your Battlesnake. If you add a path suffix or query parameters in there, the game engine will reconstruct the URL with the path and query params intact.

For example:
URL: `https://your-snake-domain.com/path/to/snake?color=green`

Metadata request: `GET https://your-snake-domain.com/path/to/snake?color=green`
Start request: `POST https://your-snake-domain.com/path/to/snake/start?color=green`
Move request: `POST https://your-snake-domain.com/path/to/snake/move?color=green`
End request: `POST https://your-snake-domain.com/path/to/snake/end?color=green`

If you're willing to have a few copies of your snake like in your second image, I think that would allow you to customize it.

**Reply by BoldBigflank** [reply]
Hey that's working really well. I am making new snakes, which is fine, but I'm passing in color/head/tail as query params so those can be custom for each snake.
